src/listpack.c

```diff
@@ -1112,6 +1112,44 @@ unsigned char *lpBatchDelete(unsigned char *lp, unsigned char **ps, unsigned lon
     return lpShrinkToFit(lp);
 }
 
+/* Trim a listpack, remove 'ltrim' elements from left and 'rtrim' elements from right. */
+unsigned char *lpTrim(unsigned char *lp, unsigned long ltrim, unsigned long rtrim) {
+    if (ltrim == 0 && rtrim == 0) return lp;
+    size_t bytes = lpBytes(lp);
+    unsigned long llen = lpLength(lp);
+    if (ltrim + rtrim >= llen) {
+        /* Remove all elements */
+        lp[LP_HDR_SIZE] = LP_EOF;
+        lpSetTotalBytes(lp, LP_HDR_SIZE + 1);
+        lpSetNumElements(lp, 0);
+        return lpShrinkToFit(lp);
+    }
+    unsigned long rangelen = llen - ltrim - rtrim;
+    unsigned char *first, *tail; /* first points to the first byte we want to save, 
+                                    and tail points to the byte after the last byte we want to save */
+    first = lpSeek(lp, ltrim);
```

Comment:
`first` may be NULL when listpack is corrupted.

```sh
config set sanitize-dump-payload no
debug set-skip-checksum-validation 1
restore _listbig 0 "\x12\x02\x02\x17\x17\x00\x00\x00\x05\x00\x82\x5F\x39\x03\x08\x01\x82\x5F\x37\x03\x06\x01\x82\x5F\x35\x03\xFF\x02\x15\x15\x00\x00\x00\x05\x75\x04\x01\x82\x5F\x33\x03\x02\x01\x82\x5F\x31\x03\x00\x01\xFF\x0B\x00\x8A\x1C\x0F\x9C\x1F\x75\x53\x82" replace
LTRIM _listbig 697 888 
LTRIM _listbig 790 702987162723 
```

You can use `./runtest --single integration/corrupt-dump-fuzzer --dont-clean --accurate` to check it.

Reply:
You are right, I have not noticed this before. And what is the right way to handle this situation? Can we simply `return lp` to terminate the trim operation when lpSeek returns null?

Reply:
we should expose that if the memory is corrupt.